csrc/executor_kernel_arg.cpp

```diff
@@ -253,6 +253,21 @@ void KernelArgumentHolder::push(const ArgAbstract* arg) {
   arguments_.emplace_back(arg->copy_unique_ptr());
 }
 
+void KernelArgumentHolder::erase(std::unique_ptr<ArgAbstract> arg) {
+  if (auto tensor_arg = dynamic_cast<TensorArgAbstract*>(arg.get())) {
+    auto iter = std::remove_if(
+        arguments_.begin(),
+        arguments_.end(),
+        [&](const std::unique_ptr<ArgAbstract>& ref) {
+          if (auto tensor_ref = dynamic_cast<TensorArgAbstract*>(ref.get())) {
+            return tensor_arg->getPointer() == tensor_ref->getPointer();
+          }
+          return false;
+        });
+    arguments_.erase(iter, arguments_.end());
```

Comment:
Are you sure this actually deallocates the tensor? It seems to me this just deallocates an `ArgAbstract`, which is our struct used as a container of an aten tensor. I thought deleting the container doesn't delete the underlying aten tensor.

Reply:
OK, I assume `at::Tensor` uses some sort of a shared pointer. We need to make sure there's no other copy except for those held in `KernelArgumentHolder`. I don't think we are keeping them anywhere else, so this seems to work.